### PR TITLE
Forsys errors and plan service cleanup

### PR DESCRIPTION
**Pull request by pflopez**
This one started as just adding some validation to forsys errors but ended up being a pretty big refactor.

Our `PlanService` is doing way too many things. Its dealing with calling backend to get plans, scenarios, and other endpoints. Its also keeping state for `planstate$`.

I've split this up into mainly 3 files
- PlanService - Backend api calls to plan related endpoints 
- ScenarioService - Backend api calls to scenario related endpoints
- PlanStateService - State related methods and other gluework

The idea behind this is to decouple state from http/backend calls, and split responsibilities between services.

While doing these services I found that we where coupling scenarios with treatmentGoalConfigs, at the service layer. Moved this upwards to the plan state service layer (there is also some jumbling of objects and data for the scenario to work downstream). In an ideal world we wouldn't have this secondary call at all, or at least not at this level to properly generate a Scenario object.
I also added some additional services, typed methods and moved things around like types.
Additionally, I've removed some artifacts about configuration, favoriteScenarios, and calls to `/plan/scores` which where not being used (and was using the deprecated `/plan/...` endpoint

So, besides refactoring stuff, this PR displays the error (if any) from creating a scenario
Screen Shot 2023-11-06 at 15 43 24

### CODE REVIEW

File: src/interface/src/app/services/errors.ts
Comment (lastminutediorama) [inline src/interface/src/app/services/errors.ts:1]: Oh nice, thanks for adding this.

File: src/interface/src/app/services/plan-state.service.spec.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+          useValue: { getTreatmentGoalsForArea: () => of([]) },
+        },
+      ],
+    });
+    service = TestBed.inject(PlanStateService);
+  });
+
+  it('should be created', () => {
+    expect(service).toBeTruthy();
+  });
```
Comment (lastminutediorama) [inline src/interface/src/app/services/plan-state.service.spec.ts:28]: Did some of the relevant tests get folded into another spec file? Apologies if I'm somehow not seeing them.

Comment (pflopez) [reply src/interface/src/app/services/plan-state.service.spec.ts:28]: There aren't any for those methods :/ I moved the ones we have for scenario to the scenario spec file but I dont think we have tests for any of this stuff

File: src/interface/src/app/types/scenario.types.ts
Comment (lastminutediorama) [inline src/interface/src/app/types/scenario.types.ts:1]: Thanks for this, as well.

**Review by lastminutediorama — APPROVED** [review]
lgtm

File: src/interface/src/app/services/plan-state.service.spec.ts (hunk 1 above)
Comment (lastminutediorama) [reply src/interface/src/app/services/plan-state.service.spec.ts:28]: ok, no worries

**Pull request merged**